### ISSUE TITLE
Enabling OpenShift authentication fails

### ISSUE THREAD

**Issue by rafik8** [kiali/kiali#1094]
**General remarks**

**Describe the bug**
I am to enable OpenShift authentication strategy when deploying to OpenShift. after enabling it I got the following UI error without any error in Kiali logs.
![image](https://user-images.githubusercontent.com/13514337/58335392-6d177400-7e39-11e9-95c6-32733d2770b0.png)

logs:
```console
I0524 14:30:51.544560       1 kiali.go:67] Kiali: Version: v0.16.2, Commit: dd6d04bd486ef1d56d0567f7cd8adcee31196867
I0524 14:30:51.544653       1 kiali.go:68] DEBUG: Kiali: Command line: [/opt/kiali/kiali -config /kiali-configuration/config.yaml -v 4]
I0524 14:30:51.544669       1 config.go:411] DEBUG: Reading YAML config from [/kiali-configuration/config.yaml]
I0524 14:30:51.545051       1 kiali.go:194] Using authentication strategy [openshift]
I0524 14:30:51.545108       1 kiali.go:88] Kiali: Console version: 0.16.0
E0524 14:30:51.553723       1 kiali.go:106] Jaeger is not available : Get http://jaeger-query/api/services:  dial tcp: lookup jaeger-query on 10.100.26.33:53: no such host
I0524 14:30:51.557929       1 server.go:49] Server endpoint will start at [:20001/]
I0524 14:30:51.557955       1 server.go:50] Server endpoint will serve static content from [/opt/kiali/console]
I0524 14:30:51.557965       1 metrics_server.go:18] Starting Metrics Server on [:9090]
I0524 14:30:51.558124       1 server.go:55] Server endpoint will require https
I0524 14:34:11.827557       1 client.go:212] DEBUG: Rest perf config QPS: 175.000000 Burst: 200
```
The deployment working fine when switch to login or anonymous option.
**Versions used**
Kiali: v0.16
Istio: 1.1.6
Kubernetes flavour and version: (e.g. OpenShift Origin 3.9):
```shell 
$ oc version
oc v3.11.0+0cbc58b
kubernetes v1.11.0+d4cacc0
features: Basic-Auth
```
**To Reproduce**
1. Export configurations:
  ```shell
   export ISTIO_NAMESPACE=istio-system-11
   export NAMESPACE=kiali-system
   export IMAGE_VERSION=v0.16
   export AUTH_STRATEGY=openshift
   export VERBOSE_MODE=4
   ```
2. Create Kiali namespace:
```shell
oc new-project kiali-system
```
3. Deploy Kiali:

```shell
curl -L https://raw.githubusercontent.com/kiali/kiali/v0.16/deploy/openshift/deploy-kiali-to-openshift.sh | sh
```

4. Invoke kiali and try to connect :
https://kiali-kiali-system.cluster.domain

4. See the error mentioned earlier.

**Expected behavior**
OpenShift will able to authenticate the user using the OAuthClient.

**Comment by jmazzitelli** [comment]
cc @mwringe @cfcosta

**Comment by israel-hdez** [comment]
@rafik8 If you go to the networking tab of  the browser "developer console", and inspect the failed request, does it have any better message the error?

**Comment by rafik8** [comment]
Here we go:
I am getting 500 Internal Server Error when heating: https://kiali-kiali-system.cluster.domain/api/auth/info

![image](https://user-images.githubusercontent.com/13514337/58342138-cf777100-7e47-11e9-9ab0-8b6c85ad6a81.png)

**Comment by israel-hdez** [comment]
@rafik8 Ok, in the lower right of your screenshot, I can see "content-type: application/json". So, at least we know the server is replying with JSON :)
Probably there is a message about the error, please check what is available in the "Response" tab:

![image](https://user-images.githubusercontent.com/23639005/58344187-a68cb700-7e1a-11e9-9e61-1d6e1c4e01f4.png)

The response from the request will bring us a better idea what to look for.

**Comment by rafik8** [comment]
here is what I get: 
![image](https://user-images.githubusercontent.com/13514337/58346316-82e56300-7e52-11e9-8ba2-5dc97242e51a.png)

**Comment by mwringe** [comment]
Can you try updating to a new version of Kiali? We had some issues with an older version where the errors could get not reported.

What version of OpenShift are you running?

**Comment by israel-hdez** [comment]
@rafik8 Thanks for your help!
From your provided _Steps to reproduce_ in your first message, I see that you are deploying kiali in _kiali-system_ namespace, while istio is installed in _istio-system-11_ namespace.

I did a quick walk through the code involved in that message and this seems to be a bug in back-end side, because a piece of code is assuming that Kiali is installed in the same namespace as Istio:

https://github.com/kiali/kiali/blob/90a16bbfa61e0b2f07aabefd581f09bb5af32a60/business/openshift_oauth.go#L166-L179

(Problematic lines seems to be 166 and 175)

Do you have the possibility to install Kiali in the same namespace as Istio? This will let us confirm if this is the issue and, at the same time, would let you use Kiali while we roll a fix.

**Comment by rafik8** [comment]
The OpenShift version is: v3.11.0.
I am using v0.16 because this is what Istio uses actually. In addition, starting from v0.18 Kiali use operator and I don't want make a lot of changes to my install based on Istio helm chart. 
That's why I have tried to separately install Kiali to confirm that the issue doesn't come from Istio Helm chart.
I will try to install the last version using the operator and let you know.

**Comment by rafik8** [comment]
@israel-hdez, humm. look interesting. I will install to istio-system-11 and update you.

**Comment by rafik8** [comment]
@israel-hdez @mwringe, I deployed Kiali to istio namespace (istio-system-11) and the Oauth integration is working well. So I could confirm the assumption of @israel-hdez.

**Comment by israel-hdez** [comment]
@rafik8 The fix was merged in master and will be available on v0.21
It probably won't be backported to v0.16 :(

**Comment by israel-hdez** [comment]
@rafik8, I'm closing this issue, since a fix is applied in `master` and will be present in Friday's release.
Please, feel free to reopen this issue if you need further help.
Thanks for your report!
